lib/WP_Auth0_Email_Verification.php

```diff
@@ -20,30 +20,36 @@ public static function init() {
 	 * @param object $userinfo
 	 */
 	public static function render_die( $userinfo ) {
+		$user_id = isset( $userinfo->user_id ) ? $userinfo->user_id : $userinfo->sub;
 
-		$html = sprintf(
-			'<p>%s<br><a id="js-a0-resend-verification" href="#">%s</a></p>
+		$html = sprintf( '<p>%s</p>', __( 'This site requires a verified email address. ', 'wp-auth0' ) );
+
+		// Only provide resend verification link for DB connection users
+		if ( 0 === strpos( $user_id, 'auth0|' ) ) {
+			$html .= sprintf(
+				'<p><a id="js-a0-resend-verification" href="#">%s</a></p>
 			<p><a href="%s?%d">%s</a></p>
 			<script>var WPAuth0EmailVerification={ajaxUrl:"%s",sub:"%s",nonce:"%s",e_msg:"%s",s_msg:"%s"}</script>
 			<script src="%s"></script>
 			<script src="%s"></script>',
-
-			__( 'Please verify your email and log in again.', 'wp-auth0' ),
-			__( 'Resend verification email.', 'wp-auth0' ),
-			wp_login_url(),
-			time(),
-			__( '← Login', 'wp-auth0' ),
-			esc_url( admin_url( 'admin-ajax.php' ) ),
-			esc_js( isset( $userinfo->user_id ) ? $userinfo->user_id : $userinfo->sub ),
-			esc_js( wp_create_nonce( self::RESEND_NONCE_ACTION ) ),
-			esc_js( __( 'Something went wrong; please login and try again.', 'wp-auth0' ) ),
-			esc_js( __( 'Email successfully re-sent to ' . $userinfo->email . '!', 'wp-auth0' ) ),
-			'//code.jquery.com/jquery-1.12.4.js',
-			WPA0_PLUGIN_URL . 'assets/js/die-with-verify-email.js?ver=' . WPA0_VERSION
-		);
+				__( 'Resend verification email.', 'wp-auth0' ),
+				wp_login_url(),
+				time(),
+				__( '← Login', 'wp-auth0' ),
+				esc_url( admin_url( 'admin-ajax.php' ) ),
+				esc_js( $user_id ),
+				esc_js( wp_create_nonce( self::RESEND_NONCE_ACTION ) ),
```

Comment:
I guess you're using the [jobs re-send email endpoint](https://auth0.com/docs/api/management/v2#!/Jobs/post_verification_email). The request only requires `user_id` and `client_id` properties. What is the nonce for?

Reply:
@lbalmaceda - WP thing to make sure that POSTed data is coming from the site. It's validated in the AJAX callback.